internal/resource/opener.go

```diff
@@ -349,8 +345,6 @@ func (ro ResourceOpener) store(
 			Fingerprint:     fingerprint,
 			RetrievedBy:     ro.retrievedBy,
 			RetrievedByType: ro.retrievedByType,
-			Origin:          origin,
```

Comment:
I think we are also fine to not double check the origin and revision here, we are asking for the resource in the database from charmhub and setting it using the same UUID directly.